lib/migration/migrator.rb

```diff
@@ -73,12 +74,28 @@ def assert_table_exists(table_name)
         fail TableNotFound, Color.red("Table[:#{table_name}] does not exist")
     end
 
+    def assert_table_does_not_exist(table_name)
+      table = @tables[Table.normalize_key(table_name)] and
+        fail DuplicateTable, Color.red("Table[:#{table_name}] already exists")
+    end
+
     def migrate(*table_names, &migration)
       table_names.each do |table_name|
         table = assert_table_exists(table_name)
         @changesets << ChangeSet.run(table, @file, &migration)
       end
       self
     end
+
+    def create_table(table_name, kind: 'type', keys:[])
```

Comment:
Looking for feedback on this API if anyone has opinions. Basic setup here is a single required positional param for the new table name, and some optional keyword params to override reasonable defaults if desired.

Reply:
Another alternative I considered that would be a little more legwork to implement but feels more DSL-ish.

```ruby
create_table(:table_name) do
  kind 'type|sellable'
  create_key(:name)

  # in theory we could also support insert/etc here but I prefer doing that in a separate migration
end
```

Reply:
I prefer the inline definitions so it's 1 method encapsulating it all, as it stands i think it's perfect the way it is.